server/SocketManager/socketmanager.js

```diff
@@ -8,4 +10,37 @@ const removeSocket = usertoken => {
   delete sockets[usertoken];
 };
 
-export { sockets, updateSocket, removeSocket };
+const emitMessage = (socket, message, data) => {
+  if (socket) {
+    socket.emit(message, data);
+  }
+};
+
+export default function initializeSocket(server) {
+  const io = new Server(server, {
+    path: '/hv/socket',
+    cors: {
+      origin: '*',
+    },
+  });
+
+  // Listen for when the client connects via socket.io-client
+  io.on('connection', socket => {
+    console.warn(`User connected ${socket.id}`);
+
+    socket.on('addUser', user => {
+      let usertoken = user.socket_id;
+      updateSocket(usertoken, socket);
+      socket.emit('AddedUser', {
+        user_added: true,
+      });
+    });
+
+    // When disconnect from client
```

Comment:
The disconnect event is currently not logging any information. It would be beneficial to log the socket ID of the user who disconnected for better debugging and monitoring.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*